editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -1187,6 +1188,13 @@ void SpriteFramesEditor::_zoom_reset() {
 }
 
 void SpriteFramesEditor::_update_library(bool p_skip_selector) {
+	if (!p_skip_selector) {
+		animations_dirty = true;
+	}
+	callable_mp(this, &SpriteFramesEditor::_update_library_impl).call_deferred();
```

Comment:
Why did you make every update_library call deferred if the problem happens only in one case?

Reply:
The code here is indeed problematic.

`_update_library()` seems to just read data from `frames` to update the GUI, calling it multiple times in a single frame can be reduced to just calling it once.